app/models/user.rb

```diff
@@ -36,11 +38,34 @@ def remember
     update_attribute :remember_digest, User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false if digest.nil?
```

Comment:
```suggestion
    return false if digest.blank?
```
please research the difference of 'blank' between 'nil'